src/main/java/org/sonarlint/intellij/analysis/SonarLintTask.java

```diff
@@ -92,50 +102,74 @@ public SonarLintJob getJob() {
 
   @Override
   public void run(ProgressIndicator indicator) {
-    AccumulatorIssueListener listener = new AccumulatorIssueListener();
+    IssueProcessor processor = SonarLintUtils.getService(myProject, IssueProcessor.class);
 
-    try {
+    Collection<Issue> issues = new ArrayList<>();
+    IssueStreamingIssueListener issueListener = new IssueStreamingIssueListener((issue -> {
+      issues.add(issue);
+      indicator.setText("Updating SonarLint issue: " + issue.getRuleKey());
       checkCanceled(indicator, myProject);
+      runInEdt(ModalityState.any(), () -> {
+        processor.process(issue);
+        myProject.getMessageBus().syncPublisher(IssueStoreListener.SONARLINT_ISSUE_STORE_TOPIC).allChanged();
+        return Unit.INSTANCE;
```

Comment:
I'm a bit concerned about having Kotlin SDK classes used in Java code. In Java you can simply return `null`